ee/tabby-webserver/src/service/repository.rs

```diff
@@ -1,13 +1,25 @@
+use anyhow::anyhow;
 use async_trait::async_trait;
 use juniper::ID;
+use tabby_common::SourceFile;
 use tabby_db::DbConn;
 
 use super::{graphql_pagination_to_filter, AsID, AsRowid};
 use crate::schema::{
-    repository::{Repository, RepositoryService},
+    repository::{FileEntry, Repository, RepositoryMeta, RepositoryService},
     Result,
 };
 
+fn glob_matches(glob: &str, mut input: &str) -> bool {
+    for part in glob.split('*') {
+        let Some((_, rest)) = input.split_once(part) else {
```

Comment:
Changing the matching algorithm to match non-contiguous substrings might be better? The effect is as shown in the following image.
<img width="720" alt="image" src="https://github.com/TabbyML/tabby/assets/17516233/43ef5f3a-5cb1-453a-abff-c61b78c0d520">

Reply:
https://github.com/TabbyML/tabby/pull/1731#discussion_r1540224477

Reply:
This is the behavior of the current algorithm if you substitute `*` for spaces. I can adjust it to split on spaces instead.